starter-code/src/functions-and-arrays.js

```diff
@@ -11,15 +17,44 @@ var words = [
   'crackpot'
 ];
 
+function findLongestWord(words) {
+  var longest=[];
+  for (i = 0; i < words.length; i++) {
+    if (longest.length < words[i].length) {
+      return longest = words[i];
+    } else if (longest.length == words[i].length){
+      return longest = words[i];
+    }
+  }
+}
+
```

Comment:
Here the else if is unnecessary. You shouldn't return `longest = words[i];` because the return is going to stop the loop. Just delete the return and it's good. 
Then you will have to return you variable `longest` outside the `for loop`.
And to pass all the test, you should add, at the beginning an `if statement` in case the array is empty.

Here is the solution: 
```
 function findLongestWord(words) {
  var longest="";
  if (words.length === 0) {
    return undefined;
  }
  for (var i = 0; i < words.length; i++) {
    if (longest.length < words[i].length) {
      longest = words[i];
    } 
  }
  return longest;
}
```